desktop/core/src/desktop/js/ko/components/ko.shareGistModal.js

```diff
@@ -32,21 +34,64 @@ const TEMPLATE = `
   </div>
   <div class="modal-body">
     <div class="row-fluid">
-      <div class="span12">
+      <div class="input-append">
         <form autocomplete="off">
-          <input id="gistLink" class="input-xxlarge" ${ window.PREVENT_AUTOFILL_INPUT_ATTRS } onfocus="this.select()" data-bind="value: link" type="text" placeholder="${ I18n('Link') }"/>
+          <input id="gistLink" style="width: 510px" ${ window.PREVENT_AUTOFILL_INPUT_ATTRS } onfocus="this.select()" data-bind="value: link" type="text" placeholder="${ I18n('Link') }"/>
+          <button class="btn" type="button" data-dismiss="modal" data-clipboard-target="#gistLink" data-bind="clipboard"><i class="fa fa-clipboard"></i></button>
         </form>
-        <button class="btn" type="button" data-dismiss="modal" data-clipboard-target="#gistLink" data-bind="clipboard">
-          <i class="fa fa-clipboard"></i> ${ I18n('Copy') }
-        </button>
       </div>
     </div>
+    <!-- ko if: window.SHARE_TO_SLACK -->
+    <label class="checkbox"><input type="checkbox" data-bind="checked: showSlackSection"> ${ I18n('Directly share to Slack') } </input></label>
+    <!-- ko if: showSlackSection -->
+    <form class="form-horizontal">
+      <div class="control-group">
+        <label class="control-label">${I18n('Message body')}</label>
+        <div class="controls">
+          <input type="text" class="input-xlarge" data-bind="value: messageDescription, valueUpdate:'afterkeydown'" placeholder="${ I18n('(Optional)') }"/>
+        </div>
+      </div>
+      <div class="control-group">
+        <label class="control-label">${I18n('Slack Channel')}</label>
+        <div class="controls">
+          <select class="input-xlarge" data-bind="options: channels, optionsCaption: 'Choose...', value: selectedChannel"></select>
+        </div>
+      </div>
+    </form>
+    <button class="btn" type="button" data-dismiss="modal" data-bind="click: postMessage">${I18n('Post Message')}</button>
```

Comment:
`Send`

Also possible to align button with message and channel?

Reply:
Made it inline-right with dropdown channel and message input length is same now @romainr 

<img width="643" alt="Screenshot 2021-04-30 at 4 35 48 PM" src="https://user-images.githubusercontent.com/42064744/116686975-21c95100-a9d2-11eb-8715-4b838d25b0e1.png">